warehouse/query-core/src/main/java/datawave/query/transformer/ExcerptTransform.java

```diff
@@ -319,51 +337,70 @@ private Set<Excerpt> getExcerpts(PhraseIndexes phraseIndexes) {
-            if (hitTermValues.contains(phrasePart)) {
-                hitPhrase.add("[" + phrasePart + "]");
+            if (i == 0) {
+                excerptIteratorOptions.put(TermFrequencyExcerptIterator.END_OFFSET, String.valueOf(end));
             } else {
-                hitPhrase.add(phrasePart);
+                // set the end of the range
+                excerptIteratorOptions.put(TermFrequencyExcerptIterator.END_OFFSET, String.valueOf(end + expandSize));
             }
-        }
+            if (log.isDebugEnabled() && i != 0) {
+                log.debug("size of excerpt requested: {}", excerptFields.getOffset(field) * 2);
+                log.debug("original range is ({},{}) and the expanded range is ({},{})", start, end,
+                                excerptIteratorOptions.get(TermFrequencyExcerptIterator.START_OFFSET),
+                                excerptIteratorOptions.get(TermFrequencyExcerptIterator.END_OFFSET));
+            }
+            try {
+                excerptIterator.init(source, excerptIteratorOptions, env);
+                excerptIterator.setHitTermsList(hitTermValues);
+                excerptIterator.setDirection(excerptFields.getDirection(field).toUpperCase().trim());
+                excerptIterator.setOrigHalfSize(origHalfSize);
+                if (i != 0) {
+                    excerptIterator.setTrim(true);
+                }
+                excerptIterator.seek(range, Collections.emptyList(), false);
+                // if an excerpt is returned...
+                if (excerptIterator.hasTop()) {
+                    Key topKey = excerptIterator.getTopKey();
+                    String[] parts = topKey.getColumnQualifier().toString().split(Constants.NULL);
+                    // The column qualifier is expected to be field\0phraseWithScores\0phraseWithoutScores\0oneBestExcerpt.
+                    if (parts.length != 4) {
+                        log.warn("{} returned top key with incorrectly-formatted column qualifier in key: {} when scanning for excerpt [{},{}] for field {} within range {}",
+                                        TermFrequencyExcerptIterator.class.getSimpleName(), topKey, start, end, field, range);
+                        return null;
+                    }
 
-        // return phrase based on direction
-        String result = String.join(" ", hitPhrase); // if no direction given, return everything
-        String direction = excerptFields.getDirection(phraseParts[0]).toUpperCase().trim();
-        if (direction.equals(BEFORE)) { // remove tokens prior to hit term
-            result = StringUtils.substringBeforeLast(result, "]") + "]";
-        } else if (direction.equals(AFTER)) { // remove tokens after hit term
-            result = "[" + StringUtils.substringAfter(result, "[");
+                    // if we have reached the limit of times to try, or we have no stop words removed
+                    if (i == 1 || !parts[1].equals("XXXWESKIPPEDAWORDXXX")) {
```

Comment:
Will we ever have a stop word removed on the first attempt (attempt 0)? If not, the '||' (or) is a little confusing here.

Reply:
my bad, let me try to explain a bit: 
If we find a stop word the first time, we will always find a stop word the second time also. The "i == 1" is so that on the second attempt, we always return what we have no matter what. The "!parts[1].equals("XXXWESKIPPEDAWORDXXX"" is so that we can return on the first attempt if there were no skip words found (basically if the iterator did not fast-fail and it returned an excerpt to us).

This part of the code is related to TermFrequencyExcerptIterator around line 340ish where if a stop word is found and this is the first iteration (know by "trim" being false), we immediately return with "XXXWESKIPPEDAWORDXXX + Constants.NULL + XXXWESKIPPEDAWORDXXX + Constants.NULL + XXXWESKIPPEDAWORDXXX" in the key. 
The "XXXWEKIPPEDAWORDXXX" is put in place of all of the excerpts to be detected here in this if statement.

Reply:
actually now that I think about it the "i ==1" might not be needed because XXXWESKIPPEDAWORDXXX can't be seen after the first attempt. I think this was another piece that was just still in here from a different implementation I tried before this one.